marlowe-playground-client/src/Halogen/ActusBlockly.purs

```diff
@@ -123,25 +137,72 @@ handleAction (GetTerms flavour) = do
   where
   unexpected s = "An unexpected error has occurred, please raise a support issue: " <> s
 
+handleAction RunAnalysis = do
+  res <-
+    runExceptT do
+      blocklyState <- ExceptT <<< map (note $ unexpected "BlocklyState not set") $ use _actusBlocklyState
+      generator <- ExceptT <<< map (note $ unexpected "Generator not set") $ use _generator
+      let
+        workspace = blocklyState.workspace
+
+        rootBlockName = blocklyState.rootBlockName
+      block <- except <<< (note $ unexpected ("Can't find root block" <> rootBlockName)) $ getBlockById workspace rootBlockName
+      except <<< lmap (\x -> "This workspace cannot be converted to code: " <> (show x)) $ blockToCode block generator
+  case res of
+    Left e -> assign _errorMessage $ Just e
+    Right contract -> do
+      assign _errorMessage Nothing
+      case parseActusJsonCode contract of 
+        Left e -> assign _errorMessage $ Just e
+        Right c -> do 
+          assign _showShiny true
+          liftEffect $ sendContractToShiny $ encodeJSON c
+      
+  where
+  unexpected s = "An unexpected error has occurred, please raise a support issue: " <> s
+
 blocklyRef :: RefLabel
 blocklyRef = RefLabel "blockly"
 
 render :: forall p. BlocklyState -> HTML p BlocklyAction
 render state =
-  div []
-    [ div
-        [ ref blocklyRef
-        , id_ "actusBlocklyWorkspace"
-        , classes [ ClassName "actus-blockly-workspace", ClassName "container-fluid" ]
-        ]
+  div [] 
+    [ section [ classes [ panelSubHeader, aHorizontal ] ]
+      [ div [ classes [ panelSubHeaderMain, aHorizontal ]]
         [ toCodeButton "Generate reactive contract"
-        , text "  "
         , toStaticCodeButton "Generate static contract"
-        , text "  (Labs is an experimental feature)"
+        -- this feature is temporary disabled because shiny is not deployed yet
+        --, runAnalysis
         , errorMessage state.errorMessage
         ]
+      ]
+    , div [ classes [ ClassName "code-panel" ]]
+      [ div
+          [ ref blocklyRef
+          , id_ "actusBlocklyWorkspace"
+          , classes [ ClassName "actus-blockly-workspace", ClassName "code-editor" ]
+          ]
+          []
+      , shiny state
+      ]
     ]
 
+shiny ::
+  forall p.
+  BlocklyState -> HTML p BlocklyAction
+shiny state =
+    aside [ classes ([ sidebarComposer, expanded false] <> if state.showShiny then [] else [hide])]
```

Comment:
since we're using a vdom it's better style to actually not render the thing rather than hide it with css

Reply:
I’ve seen things rendered through css in some places. 

Even though it’s easy to change - it’s kinda nice when you can enable disable things right from Debugger.

Reply:
that's true, and I think that's why it's seen in other places. still feels wrong though :)